app/presenters/return-logs/setup/check.presenter.js

```diff
@@ -73,6 +103,73 @@ function _alwaysRequiredPageData(session) {
   }
 }
 
+function _formatLines(lines, meter10TimesDisplay, reported, startReading, unitName) {
+  const formattedLines = []
+  let previousReading = startReading ?? 0
+
+  lines.forEach((line) => {
+    const formattedLine = {
+      endDate: new Date(line.endDate),
+      startDate: new Date(line.startDate),
+      unitName
+    }
+
+    if (reported === 'abstraction-volumes') {
+      formattedLine.quantity = line.quantity ?? null
+    } else {
+      formattedLine.reading = line.reading ?? null
+
+      if (line.reading) {
+        const multiplier = meter10TimesDisplay === 'yes' ? 10 : 1
+
+        formattedLine.quantity = (line.reading - previousReading) * multiplier
+        previousReading = line.reading
+      } else {
+        formattedLine.quantity = null
+      }
+    }
+
+    formattedLines.push(formattedLine)
+  })
+
+  return formattedLines
+}
+
+function _groupLinesByMonth(formattedLines) {
+  const groupedLines = formattedLines.reduce((acc, line) => {
+    const { endDate, quantity, reading, unitName } = line
+    const key = `${endDate.getFullYear()}-${endDate.getMonth()}`
+
+    if (!acc[key]) {
+      acc[key] = {
+        endDate,
+        quantity: 0,
```

Comment:
Might need a conversation on this one, I thought that we didn't want the lines to default to 0 as this would indicate the user has entered a reading and they've entered 0

Reply:
This is used to combine lines which are in days, weeks so it is summing the quantities. The quantities will only be null if the frequency is month